effect/ref.py

```diff
@@ -0,0 +1,60 @@
+from characteristic import attributes
+
+from ._base import Effect
+from ._dispatcher import TypeDispatcher
+from ._sync import sync_performer
+
+
+class ERef(object):
+    """
+    An effectful mutable variable.
+
+    Compare to Haskell's ``IORef`` or Clojure's ``atom``.
```

Comment:
My only suggestion is to document the purpose of this object,  e.g. from an online discussion, that it's just a way to compose code purely, to indicate that a mutable variable is being changed.  Mostly because I was confused as to what purposes this served, since from reading atom's documentation it seems like `swap!` does something more than just changing a variable in place (I guess handles threading concurrency maybe?)

Not everyone is familiar with IORef or atom.  :)